### PR TITLE
Update version of XEP-0313: Message Archive Management pt. 2

### PR DESCRIPTION
**Pull request by jacekwegr**
This PR introduces the ability to fetch specific messages by using the `ids` field. Additionally, it includes the advertisement of the `urn:xmpp:mam:2#extended` feature, as all the features required by it are now supported.

This is a follow-up to an earlier pull request: https://github.com/esl/MongooseIM/pull/4218.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/207780) / elasticsearch_and_cassandra_mnesia / be046b7564f592718086f1ad2569e4d08618963c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4225/207780/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4225/207780/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%408c802e5afd9e.2024-02-19_12.44.32/index.html)
**OK: 429** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/207887) / elasticsearch_and_cassandra_mnesia / 456f6d37ea3bf33253515069a5ce428de2a9ded5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4225/207887/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4225/207887/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ff8ac41aeb63.2024-02-20_09.06.15/index.html)
**OK: 429** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/207943) / elasticsearch_and_cassandra_mnesia / 2ad8096cc666f050301c244c08ffb31ac3756be5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4225/207943/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4225/207943/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40e8e6b58f633f.2024-02-20_09.44.23/index.html)
**OK: 429** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/208013) / elasticsearch_and_cassandra_mnesia / 0dfe8c9ae883d87e152bbbd92c3dab4e278d30f3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4225/208013/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4225/208013/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40711f76814996.2024-02-20_12.09.59/index.html)
**OK: 429** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/208046) / elasticsearch_and_cassandra_mnesia / 2b8e96f656ffd407e015bb8c3b106e94174c22cc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4225/208046/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4225/208046/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40394623adea59.2024-02-20_12.46.40/index.html)
**OK: 429** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

File: src/mam/mam_iq.erl [hunk 1]
```diff
@@ -40,6 +40,8 @@
         search_text => binary() | undefined,
         %% Filtering Result Set based on message ids
```
Comment (chrzaszcz) [inline src/mam/mam_iq.erl:41]: I think this comment is now too similar to the next one.
```suggestion
        %% Filtering Result Set before/after specific message ids
```

**Review by chrzaszcz — COMMENTED** [review]
It looks good in general. I added a few comments, and I think that the ones in `mongoose_data_forms` are the most important.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/208829) / elasticsearch_and_cassandra_mnesia / a1c80083017de6f737ff8317ca9b207c027f617a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4225/208829/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4225/208829/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%4096e0dd4b736f.2024-02-26_12.05.16/index.html)
**OK: 429** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

**Review by chrzaszcz — COMMENTED** [review]
Looks good 👍

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/209372) / elasticsearch_and_cassandra_mnesia / db4ed10f214e4840c30c1a39f43c9339dbacbd18
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4225/209372/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4225/209372/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%408c155ce3b8b5.2024-02-27_11.16.27/index.html)
**OK: 429** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]
Looks good!

**Pull request merged**